azurerm/internal/services/azurestackhci/tests/hyper_converged_cluster_resource_test.go

```diff
@@ -0,0 +1,209 @@
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Hyper Converged Cluster not found: %s", resourceName)
+		}
+
+		id, err := parse.HyperConvergedClusterID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Hyper Converged Cluster %q does not exist", id.Name)
+			}
+
+			return fmt.Errorf("bad: Get on AzureStackHCI.ClusterClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMHyperConvergedClusterDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).AzureStackHCI.ClusterClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_hyper_converged_cluster" {
+			continue
+		}
+
+		id, err := parse.HyperConvergedClusterID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on AzureStackHCI.ClusterClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMHyperConvergedCluster_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-AzureStackHCI-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func testAccAzureRMHyperConvergedCluster_basic(data acceptance.TestData) string {
+	template := testAccAzureRMHyperConvergedCluster_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_hyper_converged_cluster" "test" {
+  name                = "acctest-HyperConvergedCluster-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  client_id           = data.azurerm_client_config.current.client_id
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMHyperConvergedCluster_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMHyperConvergedCluster_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_hyper_converged_cluster" "import" {
+  name                = azurerm_hyper_converged_cluster.test.name
+  resource_group_name = azurerm_hyper_converged_cluster.test.resource_group_name
+  location            = azurerm_hyper_converged_cluster.test.location
+  client_id           = azurerm_hyper_converged_cluster.test.client_id
+  tenant_id           = azurerm_hyper_converged_cluster.test.tenant_id
+}
+`, config)
+}
+
+func testAccAzureRMHyperConvergedCluster_complete(data acceptance.TestData) string {
```

Comment:
Could we add another test case to test the update on `tags`?

Reply:
updated